x-pack/legacy/plugins/apm/server/lib/settings/agent_configuration/create_agent_config_index.ts

```diff
@@ -7,13 +7,16 @@
 import { InternalCoreSetup } from 'src/core/server';
 import { CallCluster } from '../../../../../../../../src/legacy/core_plugins/elasticsearch';
 import { getApmIndices } from '../apm_indices/get_apm_indices';
+import { getSavedObjectsClient } from '../../helpers/saved_objects_client';
 
 export async function createApmAgentConfigurationIndex(
   core: InternalCoreSetup
 ) {
   try {
     const { server } = core.http;
-    const indices = await getApmIndices(server);
+    const config = server.config();
+    const savedObjectsClient = getSavedObjectsClient(server);
```

Comment:
What do you think about renaming this to:
```suggestion
    const unscopedSavedObjectsClient = getUnscopedSavedObjectsClient(server);
```

Or perhaps:
```suggestion
    const internalSavedObjectsClient = getInternalSavedObjectsClient(server);
```

Reply:
Btw. just like you can do `server.savedObjects.getScopedSavedObjectsClient`, is it also possible to do something like: `server.savedObjects.getUnScopedSavedObjectsClient` so we don't have to init our own?

Reply:
I like the idea of making this more obvious, of the two options `internalSavedObjectsClient` would be more consistent with how we refer to it in other places.

Reply:
Thanks for chiming in @rudolf! We'll stick to `internalSavedObjectsClient` then. 
Btw. do you know if an instance of `internalSavedObjectsClient` already exists or do we need to instantiate our own?

Reply:
you need to create your own like here, what APM is doing isn't wrong, but is unusual for Kibana plugins so there's no convenience factory for that. In the NP we will probably expose an `internalRepository` but not a `internalSavedObjectsClient` (they're almost the same, the repository just has a few more methods). The idea is to make it clear that you're doing something unusual and unless you have a special reason like a legacy architecture that doesn't support spaces or writing/reading telemetry data, you probably should use the scoped saved objects client.